py/picca/pk1d/prep_pk1d.py

```diff
@@ -105,60 +109,90 @@ def exp_diff_desi(hdul, mask_targetid):
     diff: array of float
     The difference between exposures
     """
-    argsort = np.flip(np.argsort(hdul["TEFF_LYA"][mask_targetid][:]))
-    flux = hdul["FL"][mask_targetid][argsort, :]
-    ivar = hdul["IV"][mask_targetid][argsort, :]
-    teff_lya = hdul["TEFF_LYA"][mask_targetid][argsort]
-
-    num_exp = len(flux)
-    if num_exp < 2:
-        userprint("DBG : not enough exposures for diff, spectra rejected")
-        return None
-    flux_total_odd = np.zeros(flux.shape[1])
-    ivar_total_odd = np.zeros(flux.shape[1])
-    flux_total_even = np.zeros(flux.shape[1])
-    ivar_total_even = np.zeros(flux.shape[1])
-    teff_even = 0
-    teff_odd = 0
-    teff_total = np.sum(teff_lya)
-    teff_last = teff_lya[-1]
-    for index_exp in range(2 * (num_exp // 2)):
-        flexp = flux[index_exp]
-        ivexp = ivar[index_exp]
-        teff_lya_exp = teff_lya[index_exp]
-        if index_exp % 2 == 1:
-            flux_total_odd += flexp * ivexp
-            ivar_total_odd += ivexp
-            teff_odd += teff_lya_exp
-        else:
-            flux_total_even += flexp * ivexp
-            ivar_total_even += ivexp
-            teff_even += teff_lya_exp
+    argsort = np.flip(np.argsort(np.mean(file["IV"][mask_targetid], axis=1)))
 
-    w = ivar_total_odd > 0
-    flux_total_odd[w] /= ivar_total_odd[w]
-    w = ivar_total_even > 0
-    flux_total_even[w] /= ivar_total_even[w]
+    ivar_mean = np.mean(file["IV"][mask_targetid][:, :], axis=1)
+    argmin_ivar = np.argmin(ivar_mean)
+    argsort = np.arange(ivar_mean.size)
+    argsort[-1], argsort[argmin_ivar] = argsort[argmin_ivar], argsort[-1]
 
-    alpha = 1
-    if num_exp % 2 == 1:
-        # Ignasi: these variables were unsused so I commented them out
-        # n_even = (num_exp - 1) // 2
-        # alpha_n_old = np.sqrt(4. * n_even * (num_exp - n_even)) / num_exp
-        # alpha_N = np.sqrt(4.*t_even*(t_exp-t_even))/t_exp
-        # alpha_c_new = np.sqrt((teff_total - teff_last) / teff_total)
-        alpha_n_new = np.sqrt(
-            (teff_total - teff_last) * (teff_total + teff_last)) / teff_total
-        alpha = alpha_n_new
-    diff = 0.5 * (flux_total_even - flux_total_odd) * alpha
+    teff_lya = file["TEFF_LYA"][mask_targetid][argsort]
+    flux = file["FL"][mask_targetid][argsort, :]
+    ivar = file["IV"][mask_targetid][argsort, :]
 
+    n_exp = len(flux)
+    if n_exp < 2:
+        print("Not enough exposures for diff, spectra rejected")
+        return None
+    if use_only_even:
+        if n_exp % 2 == 1:
+            print("Odd number of exposures discarded")
+            return None
```

Comment:
do you want to discard odd numbers of exposures completely here? Or just discard the worst exposure? Atm you're doing the former...

Reply:
we can think about those details at a later time...